src/net.cpp

```diff
@@ -2857,12 +2857,96 @@ bool CConnman::ForNode(NodeId id, std::function<bool(CNode* pnode)> func)
     for (auto&& pnode : vNodes) {
         if(pnode->GetId() == id) {
             found = pnode;
+            found->AddRef();
             break;
         }
     }
-    return found != nullptr && NodeFullyConnected(found) && func(found);
+    bool res = found != nullptr && NodeFullyConnected(found) && func(found);
+    found->Release();
+    return res;
 }
 
+void CConnman::ForEachNode(std::function<void (CNode* pnode)> func)
+{
+    std::vector<CNode*> vNodes_copy;
+    {
+        LOCK(cs_vNodes);
+        vNodes_copy.reserve(vNodes.size());
+        for (auto&& node : vNodes) {
+            node->AddRef();
+            vNodes_copy.push_back(node);
+        }
+    }
+    for (CNode* node : vNodes_copy) {
+        if (NodeFullyConnected(node))
```

Comment:
Nit: either `{` `}` around if body, or body on the same line (in multiple places).